ecommerce/enterprise/utils.py

```
@@ -70,6 +71,30 @@ def get_enterprise_customer(site, uuid):
     }
 
 
+def get_enterprise_learner(site, user):
+    """
+    Get the EnterpriseCustomerUser with a particular username if it exists;
+    otherwise, return None.
+
+    Arguments:
+        site (Site): The site which is handling the current request
+        user (User): Django User object
+
+    Returns:
+        (dict): The single EnterpriseCustomerUser structure provided by the API
+            or returns None if EnterpriseCustomerUser with given username is not found.
+    """
+    # Handle the case when user is anonymous. We need this condition as
+    # filtering enterprise learners for anonymous user will return all learners.
+    if user.is_anonymous():
+        return None
+
+    response = enterprise_api.fetch_enterprise_learner_data(site, user)
+    results = response.get('results')
+
+    return results[0] if results else None
```

This is a code smell. Rather than return a special value, you should raise an exception (e.g. `EnterpriseUserNotFound`). Otherwise, you run the risk that someone uses this method and misses the fact that a special value is returned. An exception is hard to miss.